src/data_structures/fenwick.rs

```diff
@@ -1,28 +1,60 @@
+/// Fenwick Tree
+///
+/// Solves point add and range sum query
+///
+/// # Example
+/// ```
+/// use crate::programming_team_code_rust::data_structures::fenwick::Fenwick;
+///
+/// let mut fenwick = Fenwick::new(10);
+/// fenwick.add(0, 1);
+/// fenwick.add(1, 2);
+/// fenwick.add(2, 3);
```

Comment:
imo we only need one line per function call (e.g. one line for `add`, and one for `sum`)

the point here is to show how to use it;not to test

its already heavily tested

Reply:
I'll minimize it